src/main/java/at/ac/tuwien/kr/alpha/solver/heuristics/HeapOfRelatedChoiceAtoms.java

```diff
@@ -0,0 +1,42 @@
+package at.ac.tuwien.kr.alpha.solver.heuristics;
+
+import at.ac.tuwien.kr.alpha.common.NoGood;
+import at.ac.tuwien.kr.alpha.grounder.structure.AtomChoiceRelation;
+import at.ac.tuwien.kr.alpha.solver.ChoiceManager;
+
+import static at.ac.tuwien.kr.alpha.common.Literals.atomOf;
+
+/**
+ * A heap of active choice points that uses {@link AtomChoiceRelation} for initializing activities of related choice points.
+ *
+ * Copyright (c) 2019, the Alpha Team.
+ */
+public class HeapOfRelatedChoiceAtoms extends HeapOfActiveAtoms {
```

Comment:
The main reason for this class to exists is the initialization of variables. In `HeapOfActiveAtoms` a new nogood leads to activities of literals _occuring_ in the nogood to be updated, while here, not the literals themselves are updated, but the _related_ choice atoms of those literals occurring in the nogood. This seems to be much more in line with the idea of having good values for choice atoms (and not the "useless" ordinary atoms).

Note that `HeapOfActiveChoicePoints` works in a different direction, namely by referring to related atoms when activities are incremented. For that class, atoms are related whenever they appear in the same nogood. The `VSIDSWithPhaseSaving` heuristic, however, uses the knowledge of atom relationships that it obtains from the grounder via the `AtomChoiceRelation`. (I currently do however not remember whether the obtained information in both cases is the same or different. The information flow in `VSIDSWithPhaseSaving` and `HeapOfRelatedChoiceAtoms` seems to be more direct.)

I think, it is worthwhile to consider merging both approaches. If I recall correctly, I did not do it, because there were some differences and I did not want to change all the other heuristics we have so far without any discussion.